client/app/bundles/course/reference-timelines/components/TimelinesOverview/TimelinesOverviewItem.tsx

```diff
@@ -0,0 +1,132 @@
+import { useState } from 'react';
+import { toast } from 'react-toastify';
+import { MoreVert } from '@mui/icons-material';
+import { Divider, IconButton, Menu, MenuItem } from '@mui/material';
+import { TimelineData } from 'types/course/referenceTimelines';
+
+import Checkbox from 'lib/components/core/buttons/Checkbox';
+import useTranslation from 'lib/hooks/useTranslation';
+
+import { useLastSaved, useSetLastSaved } from '../../contexts';
+import { deleteTimeline } from '../../operations';
+import { useAppDispatch } from '../../store';
+import translations from '../../translations';
+import CreateRenameTimelinePrompt from '../CreateRenameTimelinePrompt';
+import DeleteTimelinePrompt from '../DeleteTimelinePrompt';
+
+interface TimelinesOverviewItemProps {
+  for: TimelineData;
+  checked?: boolean;
+  onChangeCheck?: (checked: boolean) => void;
+}
+
+const TimelinesOverviewItem = (
+  props: TimelinesOverviewItemProps,
+): JSX.Element => {
+  const { for: timeline } = props;
+
+  const { t } = useTranslation();
+
+  const dispatch = useAppDispatch();
+
+  const [menuAnchor, setMenuAnchor] = useState<HTMLButtonElement>();
+  const [renaming, setRenaming] = useState(false);
+  const [deleting, setDeleting] = useState(false);
+
+  const { status } = useLastSaved();
+  const { startLoading, abortLoading, setLastSavedToNow } = useSetLastSaved();
+
+  const handleDelete = (alternativeTimelineId?: TimelineData['id']): void => {
+    startLoading();
+
+    dispatch(deleteTimeline(timeline.id, alternativeTimelineId))
+      .then(setLastSavedToNow)
+      .catch((error) => {
+        abortLoading();
+
+        toast.error(
+          error ??
+            t(translations.errorDeletingTimeline, { title: timeline.title }),
+        );
+      });
+  };
+
+  return (
+    <div key={timeline.id} className="flex shrink-0 items-start">
+      <Checkbox
+        checked={props.checked}
+        description={
+          !timeline.default && timeline.timesCount
+            ? t(translations.nAssigned, { n: timeline.timesCount })
+            : undefined
+        }
+        descriptionVariant="caption"
+        disabled={timeline.default ?? status === 'loading'}
+        label={timeline.title}
+        labelClassName="!-mb-2 line-clamp-1 max-w-[20rem]"
+        onChange={(_, checked): void => props.onChangeCheck?.(checked)}
+        variant="body2"
+      />
+
+      {!timeline.default && (
```

Comment:
Not sure if I get your question right, but are you asking why I did not put `Menu`, `CreateRenameTimelinePrompt`, and `DeleteTimelinePrompt` together with `IconButton` in a fragment, and render them if `!timeline.default`?

If yes, then my answer is that there is no need to. [MUI `Modal`s will not render anything to the DOM if the `open` prop is falsy](https://github.com/mui/material-ui/blob/f88330c4d81c50282f7445acd79d1cea64325c9d/packages/mui-base/src/ModalUnstyled/ModalUnstyled.tsx#L290). Nesting them inside will just increase the codes' tab count.

Plus, if we are going to add _Duplicate_ and _Offset by..._ to the menu, the Default Timeline will also need an `IconButon`.

Reply:
"Plus, if we are going to add Duplicate and Offset by... to the menu, the Default Timeline will also need an IconButon."

Okay fair if this is the case. Otherwise I was not sure why the checks for the same logic outcome were separated.